### PR TITLE
[camera] Remove OCMock from permission tests 

### PR DESCRIPTION
**Pull request by mchudy**
Smaller part extracted from https://github.com/flutter/packages/pull/8342

- Removes OCMock from `CameraPermissionTests`
- Wraps permission methods into a new interface `FLTCameraPermissionManager`
- Introduces new protocol which wraps framework methods and can be mocked directly `FLTPermissionService` 

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [X] I signed the [CLA].
- [X] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I [linked to at least one issue that this PR fixes] in the description above.
- [X] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [X] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPermissionTests.m [hunk 1]
```diff
@@ -105,21 +138,22 @@ - (void)testRequestCameraPermission_completeWithErrorIfUserDenyAccess {
...
+  };
 
-  // Mimic user choosing "deny" in permission dialog.
-  OCMStub([mockDevice requestAccessForMediaType:AVMediaTypeVideo
-                              completionHandler:[OCMArg checkWithBlock:^BOOL(void (^block)(BOOL)) {
-                                block(NO);
-                                return YES;
-                              }]]);
-  FLTRequestCameraPermissionWithCompletionHandler(^(FlutterError *error) {
+  // Mimic user choosing "allow" in permission dialog.
```
Comment (hellohuanlin) [inline packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPermissionTests.m:146]: just double check - is it deny? (`handler(NO)`) below.

File: packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPermissionTests.m [hunk 2]
```diff
@@ -8,31 +8,60 @@
 #endif
 @import AVFoundation;
 @import XCTest;
-#import <OCMock/OCMock.h>
+
 #import "CameraTestUtils.h"
+#import "FLTCameraPermissionManager.h"
+#import "FLTPermissionService.h"
```
Comment (hellohuanlin) [inline packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPermissionTests.m:14]: Nit: "managing" and "servicing" for protocol naming as discussed in the [design doc](https://docs.google.com/document/d/1XsaulkJA6_ZSpM7chkQLhQY25sQqhEMiqHMYzw8H85o/edit?resourcekey=0-_cUjF1c0iBvRLKfV-3gK5A&tab=t.0#heading=h.a6hzk7tvmg6z).

**Review by hellohuanlin — APPROVED** [review]
LGTM after nits

**Review by jmagman — CHANGES_REQUESTED** [review]
Ci is complaining about https://github.com/flutter/flutter/issues/160799, can you rebase/merge past https://github.com/flutter/packages/pull/8344 ?

Additionally, this will need a version bump since there aren't just changes to tests, but also `camera_avfoundation/ios/camera_avfoundation/Sources`.
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8727393179177658609/+/u/Run_package_tests/CHANGELOG_and_version_validation/stdout

https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version-and-changelog-updates

File: packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPermissionTests.m (hunk 1 above)
Comment (mchudy) [reply packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPermissionTests.m:146]: You're right, thanks for the catch

File: packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPermissionTests.m (hunk 2 above)
Comment (mchudy) [reply packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPermissionTests.m]: I changed `FLTPermissionService` protocol name to `FLTPermissionServicing`. Not sure if `FLTCameraPermissionManager.h` should also be renamed as it is not a protocol

Comment (hellohuanlin) [reply packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPermissionTests.m]: Oh if it's not a protocol no need to change.

**Comment by mchudy** [comment]
Ok, I think I fixed everything, the CI is passing now

**Review by jmagman — APPROVED** [review]
LGTM, thank you so much!

**Pull request merged**